plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -0,0 +1,1180 @@
+                VerticalGroup:new{
+                    dimen = Geom:new{
+                        w = text_max_width,
+                        h = self.height,
+                    },
+                    self.v_spacer,
+                    LeftContainer:new{
+                        dimen = Geom:new{w = text_max_width, h = word_height},
+                        word_widget
+                    },
+                    LeftContainer:new{
+                        dimen = Geom:new{w = text_max_width, h = self.height - word_height - self.v_spacer.width*2.2},
+                        HorizontalGroup:new{
+                            subtitle_prefix,
+                            TextWidget:new{
+                                text = self.item.book_title,
+                                face = subtitle_italic_face,
+                                max_width = math.ceil(math.max(5,text_max_width - subtitle_prefix:getSize().w - Size.padding.fullscreen)),
+                                fgcolor = subtitle_color
+                            }
+                        }
+                    },
+                    self.v_spacer
+                }
+
+            },
+            RightContainer:new{
+                dimen = Geom:new{ w = right_side_width+Size.padding.default, h = self.height},
+                right_widget
+            }
+        },
+    }
+end
+
+function VocabItemWidget:getTimeSinceDue()
+    if self.item.review_count >= 8 then return "" end
+
+    local elapsed = os.time() - self.item.due_time
+    local abs = math.abs(elapsed)
+    local readable_time
+
+    local rounding = elapsed > 0 and math.floor or math.ceil
+    if abs < 60 then
+        readable_time = abs .. "s"
+    elseif abs < 3600 then
+        readable_time = string.format("%dm", rounding(abs/60))
+    elseif abs < 3600 * 24 then
+        readable_time = string.format("%dh", rounding(abs/3600))
+    else
+        readable_time = string.format("%dd", rounding(abs/3600/24))
+    end
+
+    if elapsed < 0 then
+        return " " .. readable_time .. " | " --hourglass
+    else
+        return readable_time .. " | "
+    end
+end
+
+function VocabItemWidget:remover()
+    self.item.remove_callback(self.item)
+    self.show_parent:removeAt(self.index)
+end
+
+function VocabItemWidget:resetProgress()
+    self.item.review_count = 0
+    self.item.due_time = os.time() - 1
+    DB:resetWordProgress(self.item.word)
+    self:initItemWidget()
+    UIManager:setDirty(self.show_parent, function()
+        return "ui", self[1].dimen end)
+end
+
+function VocabItemWidget:removeAndClose()
+    self:remover()
+    UIManager:close(self.dialogue)
+end
+
+function VocabItemWidget:showMore()
+    local dialogue = WordInfoDialog:new{
+        title = self.item.word,
+        book_title = self.item.book_title,
+        dates = _("Added on ") .. os.date("%Y-%m-%d", self.item.create_time) .. " | " ..
+        _("Review scheduled at ") .. os.date("%Y-%m-%d %H:%M", self.item.due_time),
```

Comment:
The translators shouldn't be bothered with spaces.

```suggestion
        dates = _("Added on") .. " " .. os.date("%Y-%m-%d", self.item.create_time) .. " | " ..
        _("Review scheduled at") .. " " .. os.date("%Y-%m-%d %H:%M", self.item.due_time),
```

(I'll refrain from remarking something about localization and date formats; ISO-style YYYY-mm-dd should be sufficiently universal for the moment.)

Reply:
Done.